labonneboite/web/user/views.py

```diff
@@ -57,7 +57,7 @@ def account_delete():
         # We have to delete it because it has a foreign key to the User table.
         # We don't need to deal with the other tables of Social Auth, see:
         # https://python-social-auth.readthedocs.io/en/latest/storage.html
-        db_session.query(UserSocialAuth).filter_by(id=user_social_auth.id).delete()
+        db_session.query(UserSocialAuth).filter_by(user_id=current_user.id).delete()
```

Comment:
Du coup, est-il toujours nécessaire d'avoir une variable `user_social_auth` vu qu'elle n'est utilisé qu'en fin de la fonction ?

`return logout(user_social_auth=user_social_auth)` pourrait devenir `return logout(user_social_auth=get_user_social_auth(current_user.id))

Reply:
C'est une bonne idée, et j'étais sur le point d'appliquer ta recommandation mais j'ai lu le commentaire : "Store the current `UserSocialAuth` instance in memory because it will be deleted but it will also be needed later to properly logout the user from PEAM.".